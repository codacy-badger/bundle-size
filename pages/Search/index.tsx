import { styles } from '../../client/styles/SearchPageStyles';
import { useState, ChangeEvent, KeyboardEvent, useRef } from 'react';
import Link from 'next/link';

export default function Search() {
  const [searchInput, setSearch] = useState('');
  const searchSubmitRef = useRef<HTMLAnchorElement>(null);

  const onChangeSearchInput = ({ target }: ChangeEvent<HTMLInputElement>) => {
    setSearch(target.value);
  };

  const submitSearch = () => {
    searchSubmitRef.current?.click();
  };

  const onKeyPress = (event: KeyboardEvent) => {
    const { key } = event;

    if (key === 'Enter') {
      event.preventDefault();
      submitSearch();
    }
  };

  return (
    <>
      <div className="Search">
        <h1 className="Search__Title">
          Find the cost of adding a npm package to your bundle
        </h1>
        <div className="Search__Input-container">
          <form className="Search__Input-form">
            <div className="Input__Container">
              <div className="Input__Container-inner">
                <input
                  placeholder="find package"
                  className="Input"
                  value={searchInput}
                  onChange={onChangeSearchInput}
                  onKeyPress={onKeyPress}
                />
              </div>
            </div>
          </form>
        </div>
      </div>

      <Link href="/package/[id]" as={`package/${searchInput}`}>
        <a ref={searchSubmitRef} />
      </Link>

      <style jsx>{styles}</style>
    </>
  );
}
