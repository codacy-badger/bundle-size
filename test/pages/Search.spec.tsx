import React from 'react';
import { render } from '../testUtils';
import Search from '../../pages/Search';

describe('Search page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Search />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
