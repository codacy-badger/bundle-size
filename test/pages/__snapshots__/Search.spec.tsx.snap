// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-2704832905 Search"
  >
    <h1
      class="jsx-2704832905 Search__Title"
    >
      Find the cost of adding a npm package to your bundle
    </h1>
    <div
      class="jsx-2704832905 Search__Input-container"
    >
      <form
        class="jsx-2704832905 Search__Input-form"
      >
        <div
          class="jsx-2704832905 Input__Container"
        >
          <div
            class="jsx-2704832905 Input__Container-inner"
          >
            <input
              class="jsx-2704832905 Input"
              placeholder="find package"
              value=""
            />
          </div>
        </div>
      </form>
    </div>
  </div>
  <a
    class="jsx-2704832905"
    href="package/"
  />
</DocumentFragment>
`;
